### PR TITLE
fix compilation warnings

### PR DESCRIPTION
**Pull request by schnek**
**Todo / Checklist**
- [x] Target is branch develop.
- [x] Checked AE-Coding standards.

**Tests Performed:** 
- [x] Build AE.
- [x] Server startup.
- [x] Log into world.

**Multiversion Ingame Tests Performed:**
- [x] TBC
- [x] WotLK
- [x] Cata

### CODE REVIEW

File: src/world/Objects/Units/Unit.cpp [hunk 1]
```diff
@@ -75,7 +75,7 @@ Unit::Unit() :
 #endif
 
     // Zyres: initialise here because multiversion differences
-    std::fill_n(PctPowerRegenModifier, TOTAL_PLAYER_POWER_TYPES, 1);
+    std::fill_n(PctPowerRegenModifier, TOTAL_PLAYER_POWER_TYPES, '1');
```
Comment (Appled) [inline src/world/Objects/Units/Unit.cpp:78]: You are initializing float values with char '1' which is 49 in integer and that would be bad... Try with 1.0f

**Review by Appled — CHANGES_REQUESTED** [review]

File: src/world/Objects/Units/Unit.cpp (hunk 1 above)
Comment (schnek) [reply src/world/Objects/Units/Unit.cpp]: up🎈

Comment (Appled) [reply src/world/Objects/Units/Unit.cpp]: Without quotes. Like this (PctPowerRegenModifier, TOTAL_PLAYER_POWER_TYPES, 1.0f);

File: src/world/Objects/Units/Unit.cpp [hunk 2]
```diff
@@ -75,7 +75,7 @@ Unit::Unit() :
 #endif
 
     // Zyres: initialise here because multiversion differences
-    std::fill_n(PctPowerRegenModifier, TOTAL_PLAYER_POWER_TYPES, 1);
+    std::fill_n(PctPowerRegenModifier, TOTAL_PLAYER_POWER_TYPES, '1.0f');
```
Comment (Appled) [inline src/world/Objects/Units/Unit.cpp:78]: Without quotes. Like this (PctPowerRegenModifier, TOTAL_PLAYER_POWER_TYPES, 1.0f);

**Review by Appled — CHANGES_REQUESTED** [review]

File: src/world/Objects/Units/Unit.cpp (hunk 2 above)
Comment (schnek) [reply src/world/Objects/Units/Unit.cpp:78]: cleared cache and compiled (without '')
https://github.com/fmtlib/fmt/blob/master/include/fmt/chrono.h#L1692 I looked at the option from 'format' there just with quotes

**Review by Appled — APPROVED** [review]

**Pull request merged**